packages/flutter/lib/src/cupertino/bottom_tab_bar.dart

```diff
@@ -175,4 +180,30 @@ class CupertinoTabBar extends StatelessWidget {
       ),
     );
   }
+
+  /// Clone the current [CupertinoTabBar] with the [currentIndex] overridden
+  /// by `withCurrentIndex` and the [onTap] chained after `withOnTap` if present.
+  CupertinoTabBar clone({
+    @required int withCurrentIndex,
+    @required ValueChanged<int> withOnTap,
+  }) {
+    assert(withCurrentIndex != null);
+    assert(withOnTap != null);
+    return new CupertinoTabBar(
+       key: key,
+       items: items,
+       backgroundColor: backgroundColor,
+       activeColor: activeColor,
+       inactiveColor: inactiveColor,
+       iconSize: iconSize,
+       // Directly override the index.
+       currentIndex: withCurrentIndex,
+       // Concatenate the 2 callback functions.
+       onTap: (int newIndex) {
+         withOnTap(newIndex);
```

Comment:
this seems dangerous... what if you keep applying it to the object you get back? Eventually you'll have this really long chain of callbacks...